src/cons/ledger_handler.cpp

```diff
@@ -0,0 +1,89 @@
+
+#include <flatbuffers/flatbuffers.h>
+#include <iostream>
+#include <fstream>
+#include <boost/filesystem.hpp>
+#include "../conf.hpp"
+#include "../crypto.hpp"
+#include "../p2p/p2p.hpp"
+#include "../fbschema/ledger_helpers.hpp"
+#include "ledger_handler.hpp"
+
+namespace cons
+{
+
+std::string save_ledger(const p2p::proposal &proposal, const uint64_t led_seq_no)
+{
+    //Serialize lcl using flatbuffer ledger schema.
+    flatbuffers::FlatBufferBuilder builder(1024);
+    const std::string_view ledger_str = fbschema::ledger::create_ledger_from_proposal(builder, proposal);
+
+    //Get binary hash of the the serialized lcl.
+    const std::string lcl = crypto::get_hash(ledger_str);
+
+    //Get hex from binary hash
+    std::string lcl_hash;
+    util::bin2hex(lcl_hash,
+                  reinterpret_cast<const unsigned char *>(lcl.data()),
+                  lcl.size());
+
+    //create file path to save lcl.
+    //file name -> [ledger sequnce numer]-lcl[lcl hex]
+    std::string path;
+    std::string seq_no = std::to_string(led_seq_no);
+    path.reserve(conf::ctx.histDir.size() + lcl_hash.size() + seq_no.size() + 5);
+    path.append(conf::ctx.histDir);
+    path.append("/");
+    path.append(seq_no);
```

Comment:
You can use `std::move(seq_no)`

Reply:
Sorry, disregard above. In this case std::move does not make a difference for the “append” operation because it anyway has to copy characters from the given string and append to the other.